atlasdb-commons/src/main/java/com/palantir/async/initializer/AsyncInitializer.java

```diff
@@ -37,10 +37,9 @@
 public abstract class AsyncInitializer {
     private static final Logger log = LoggerFactory.getLogger(AsyncInitializer.class);
 
-    private final ScheduledExecutorService singleThreadedExecutor = getExecutorService();
+    private final ScheduledExecutorService singleThreadedExecutor = createExecutorService();
     private final AtomicBoolean isInitializing = new AtomicBoolean(false);
-    private volatile boolean initialized = false;
-    private volatile boolean canceledInitialization = false;
+    private volatile AsyncInitializationState state = new AsyncInitializationState();
```

Comment:
why is this volatile? you never update it

Reply:
Relic from first impl